controllers/chaosimpl/physicalmachinechaos/impl.go

```diff
@@ -119,51 +116,48 @@ func (impl *Impl) Recover(ctx context.Context, index int, records []*v1alpha1.Re
 	impl.Log.Info("recover physical machine chaos")
 
 	physicalMachinechaos := obj.(*v1alpha1.PhysicalMachineChaos)
-	addresses := records[index].Id
-
-	addressArray := strings.Split(addresses, ",")
-	for _, address := range addressArray {
-		url := fmt.Sprintf("%s/api/attack/%s", address, physicalMachinechaos.Spec.ExpInfo.UID)
-		statusCode, err := impl.doHttpRequest("DELETE", url, nil)
-		if err != nil {
-			return v1alpha1.Injected, err
-		}
-
-		if statusCode == http.StatusNotFound {
-			impl.Log.Info("experiment not found", "uid", physicalMachinechaos.Spec.ExpInfo.UID)
-		} else if statusCode != http.StatusOK {
-			err = errors.New("HTTP status is not OK")
-			impl.Log.Error(err, "")
-			return v1alpha1.Injected, err
-		}
+	address := records[index].Id
+
+	url := fmt.Sprintf("%s/api/attack/%s", address, physicalMachinechaos.Spec.ExpInfo.UID)
+	statusCode, body, err := impl.doHttpRequest("DELETE", url, nil)
+	if err != nil {
+		return v1alpha1.Injected, errors.Wrap(err, body)
+	}
+
+	if statusCode == http.StatusNotFound {
+		impl.Log.Info("experiment not found", "uid", physicalMachinechaos.Spec.ExpInfo.UID)
+	} else if statusCode != http.StatusOK {
+		err = errors.New("HTTP status is not OK")
+		impl.Log.Error(err, body)
+		return v1alpha1.Injected, errors.Wrap(err, body)
 	}
 
 	return v1alpha1.NotInjected, nil
 }
 
-func (impl *Impl) doHttpRequest(method, url string, data io.Reader) (int, error) {
+func (impl *Impl) doHttpRequest(method, url string, data io.Reader) (int, string, error) {
 	req, err := http.NewRequest(method, url, data)
 	if err != nil {
 		impl.Log.Error(err, "fail to generate HTTP request")
-		return 0, err
+		return 0, "", err
 	}
 	req.Header.Set("Content-Type", "application/json")
 
 	httpClient := &http.Client{}
 	resp, err := httpClient.Do(req)
 	if err != nil {
 		impl.Log.Error(err, "do HTTP request")
-		return 0, err
+		return 0, "", err
 	}
 	defer resp.Body.Close()
 
 	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
As we are using go1.16 now, and [this document](https://golang.org/doc/go1.16#ioutil) suggests us to use `io.ReadAll` instead of `ioutil.ReadAll`.

Reply:
done